### ISSUE TITLE
Consider making trace reference more general in the IR

### ISSUE THREAD

**Issue by bobbinth** [0xPolygonMiden/air-script#54]
Currently, in the [Operation](https://github.com/0xPolygonMiden/air-dsl/blob/main/ir/src/transition_constraints/graph.rs#L245) enum the way we reference trace columns is via `MainTraceCurrentRow` and `MainTraceNextRow` (and similar for auxiliary trace). This hard-codes the evaluation frame to work only with two consecutive rows. Instead, we could make the structure more general by putting the row offset into the enum itself. For example, the enum could look like this:

```Rust
pub enum Operation {
    Const(u64),
    MainTraceRow(usize, usize),
    AuxTraceRow(usize, usize),
    ...
}
```
In the `MainTraceRow` and `AuxTraceRow` the first `usize` would refer to the column offset and the second `usize` would refer to the row offset. So, for example: `MainTraceRow(1, 0)` would mean column 1 at the current row, but `MainTraceRow(1, 1)` would mean column 1 at the next row.

**Comment by grjte** [comment]
I like this overall, and it seems like the priority is bumped up by #84.

One thought I have thought is that I don't love having the offset data as usizes where order matters. I wonder if maybe instead we want some kind of light struct that has the Trace row information, where we can also move the information about which trace it's in. This would also make things more flexible if we wanted more than just Main and Auxiliary traces in the future.

Here's a rough idea, though the names could be improved:

```
pub enum Operation {
    Const(u64),
    TraceElement(TraceAccess), // replaces `MainTraceRow` and `AuxTraceRow`
    ...
}


struct TraceAccess {
    trace: TraceType
    col_idx: usize,
    row_offset: usize,
}

pub enum TraceType {
    Main,
    Aux
}
```

**Comment by grjte** [comment]
Another thought - to make it easy to expand beyond main and aux traces, we could identify the trace with a number indicating how many commitments came before the trace (or how many times we got fresh randomness from the verifier), e.g. our current `Main` would be trace `0`, our current `Aux` would be trace `1`, etc. Then at least the IR part of this supports more traces (although language syntax + parser would need adjustment for future changes)

Then we'd have something like:

```
struct TraceAccess {
    trace: u8,
    col_idx: usize,
    row_offset: usize,
}
```

**Comment by bobbinth** [comment]
I like the latter approach as it does not tie us into the terminology of `main` and `aux` names.

The only suggestion is that maybe we follow the naming convention of winterfell and call these `trace_segment`. So, the struct would look like this:
```Rust
struct TraceAccess {
    trace_segment: u8,
    col_idx: usize,
    row_offset: usize,
}
```

**Comment by tohrnii** [comment]
Closed by #92.
